### PR TITLE
`make test` can run in `-j` parallel mode

### LINKED ISSUE #957: Parallel test does not work
**Describe the bug**

`make check -j` fails:
image

Likely to be a race condition for filename.

**Expected behavior**

Support parallelism as other target.

**System (please complete the following information):**
 - OS: [e.g. Mac] CentOS 7
 - Version [e.g. 22] v1.9.3
 - Compiler [e.g. gcc] gcc-9 (devtoolset-9)
 - Build System [e.g. Makefile] makefile
 - Other hardware specs [e.g Core 2 duo...] Xeon Gold 6140

### PR DESCRIPTION
**Pull request by Cyan4973**
fix #957

### CODE REVIEW

File: tests/Makefile [hunk 1]
```diff
@@ -201,279 +201,300 @@ test-compile-with-lz4-memory-usage:
 	$(MAKE) clean; CFLAGS=-O0 CPPFLAGS=-D'LZ4_MEMORY_USAGE=LZ4_MEMORY_USAGE_MAX' $(MAKE) all
 
 .PHONY: test-lz4-sparse
+# Rules regarding Temporary test files :
+# Each test must use its own unique set of names during execution.
+# Each temporary test file must begin by an FPREFIX.
+# Each FPREFIX must be unique for each test.
```
Comment (t-mat) [inline tests/Makefile:207]: This is a confirmation for minor/corner case.
As for uniqueness, we also do not accept common prefix of `FPREFIX`.

For example

```
# NG
test-1: FPREFIX = tmp-test-1
	...
	@$(RM) $(FPREFIX)*  # This command (`rm tmp-test-1*`) also removes `tmp-test-1-1*`
...
test-1-1: FPREFIX = tmp-test-1-1
	...
	@$(RM) $(FPREFIX)*
...
```

**Comment by t-mat** [comment]
I've compared durations of the following tests.

- #1120 (This PR) : https://github.com/lz4/lz4/runs/7362381775?check_suite_focus=true
- #1119 (Previous PR) : https://github.com/lz4/lz4/runs/7379894072?check_suite_focus=true

And glad to see a superb improvement for all `make -C tests` variants.  Especially for `-mx32`!

**Comment by xkszltl** [comment]
Good to see the large speed up in `make test`, together with the `-j` support.
Previously we can only afford `make check` in most builds.
